ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright 2019 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.cloud;
+
+import com.github.ambry.clustermap.PartitionId;
+import com.github.ambry.commons.BlobId;
+import com.github.ambry.config.VerifiableProperties;
+import com.github.ambry.store.FindInfo;
+import com.github.ambry.store.FindToken;
+import com.github.ambry.store.MessageInfo;
+import com.github.ambry.store.MessageWriteSet;
+import com.github.ambry.store.Store;
+import com.github.ambry.store.StoreErrorCodes;
+import com.github.ambry.store.StoreException;
+import com.github.ambry.store.StoreGetOptions;
+import com.github.ambry.store.StoreInfo;
+import com.github.ambry.store.StoreKey;
+import com.github.ambry.store.StoreStats;
+import com.github.ambry.store.Write;
+import com.github.ambry.utils.ByteBufferInputStream;
+import com.github.ambry.utils.Utils;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ReadableByteChannel;
+import java.util.EnumSet;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * The blob store that reflects data in a cloud storage.
+ */
+class CloudBlobStore implements Store {
+
+  private final Logger logger = LoggerFactory.getLogger(getClass());
+  private final PartitionId partitionId;
+  private CloudDestination cloudDestination;
+  private boolean started;
+
+  /**
+   * Constructor for CloudBlobStore
+   * @param partitionId partition associated with BlobStore.
+   * @param verProps the cloud store properties.
+   */
+  CloudBlobStore(PartitionId partitionId, VerifiableProperties verProps, CloudDestination cloudDestination) {
+    this.cloudDestination = cloudDestination;
+    this.partitionId = partitionId;
+  }
+
+  @Override
+  public void start() throws StoreException {
+    started = true;
+  }
+
+  @Override
+  public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> storeGetOptions) throws StoreException {
+    throw new UnsupportedOperationException("Method not supported");
+  }
+
+  @Override
+  public void put(MessageWriteSet messageSetToWrite) throws StoreException {
+    checkStarted();
+    if (messageSetToWrite.getMessageSetInfo().isEmpty()) {
+      throw new IllegalArgumentException("Message write set cannot be empty");
+    }
+    checkDuplicates(messageSetToWrite);
+
+    // Write the blobs in the message set
+    CloudWriteChannel cloudWriter = new CloudWriteChannel(this, messageSetToWrite.getMessageSetInfo());
```

Comment:
sending `this` looks a bit suspect here. Why not pass what it needs?

Reply:
See previous comment.  Want to keep this inner class very simple and have it delegate all the logic to the Store class.